pkgs/swiftgen/swift2objc/lib/src/generator/generators/class_generator.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:swift2objc/src/ast/_core/shared/referred_type.dart';
+import 'package:swift2objc/src/ast/declarations/compounds/class_declaration.dart';
+import 'package:swift2objc/src/generator/_core/utils.dart';
+
+String generateClass(ClassDeclaration declaration) {
+  final lines = [
+    _generateClassHeader(declaration),
+    _generateClassWrappedInstance(declaration),
+    _generateClassInitializer(declaration),
+    ..._generateClassMethods(declaration),
+    "}",
+  ].nonNulls.toList();
+
+  return lines.join("\n");
+}
+
+String _generateClassHeader(ClassDeclaration declaration) {
+  var header = "";
+
+  if (declaration.hasObjCAnnotation) {
+    header = "@objc ";
+  }
+
+  header += "public class ${declaration.name}";
+
+  final superClassAndProtocols = [
+    declaration.superClass?.declaration.name,
+    ...declaration.conformedProtocols
+        .map((protocol) => protocol.declaration.name),
+  ];
+
+  if (superClassAndProtocols.isNotEmpty) {
+    header = "$header: ${superClassAndProtocols.join(", ")}";
+  }
+
+  header += " {";
+
+  return header;
+}
+
+String? _generateClassWrappedInstance(ClassDeclaration declaration) {
+  final property = declaration.wrappedInstance;
+
+  if (property == null) return null;
+
+  assert(
+    property.type is! GenericType,
+    "Wrapped instance can't have a generic type",
+  );
+
+  return "var ${property.name}: ${property.type.name}";
+}
+
+String? _generateClassInitializer(ClassDeclaration declaration) {
+  final initializer = declaration.initializer;
+  if (initializer == null) return null;
+
+  return [
+    "init(${generateParameters(initializer.params)}) {",
+    ...initializer.statements,
+    "}"
+  ].toList().join("\n");
+}
+
+List<String> _generateClassMethods(ClassDeclaration declaration) {
+  return declaration.methods.map(
+    (method) {
+      var header = "";
```

Comment:
Another good candidate for a multi-line string literal

Reply:
I used the StringBuffer for better performance. I just wanna keep the indentations intact until we start using the formatter.